### PR TITLE
cli: add `ui.always-allow-large-revsets` option

### PR DESCRIPTION
**Pull request by thoughtpolice**
This lets users use "large" revsets in commands such as `jj rebase`, without needing the `all:` modifier.

# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [x] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/config-schema.json [hunk 1]
```diff
@@ -334,6 +334,11 @@
                     "type": "string",
                     "description": "Revisions to give shorter change and commit IDs to",
                     "default": "<revsets.log>"
+                },
+                "always-allow-large-revsets": {
+                    "type": "boolean",
+                    "description": "Whether to allow large revsets to be used in all commands without the `all:` modifier",
```
Comment (noahmayr) [inline cli/src/config-schema.json:340]: Nit: I think large is the wrong word for it, maybe `non-singleton-revsets`?

**Review by noahmayr — COMMENTED** [review]
Not sure about the term "large revsets", apart from that lgtm

**Review by noahmayr — APPROVED** [review]

File: cli/tests/test_rebase_command.rs [hunk 2]
```diff
@@ -496,6 +498,32 @@ fn test_rebase_multiple_destinations() {
     ◉
     "###);
 
+    // undo and do it again, but with 'revsets.always-allow-large-revsets'
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["undo"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @"");
```
Comment (martinvonz) [inline cli/tests/test_rebase_command.rs:504]: nit: we can probably skip this.

File: cli/src/config/misc.toml [hunk 3]
```diff
@@ -5,6 +5,8 @@ amend = ["squash"]
 co = ["checkout"]
 unamend = ["unsquash"]
 
+[revsets]
```
Comment (martinvonz) [inline cli/src/config/misc.toml:8]: We have talked about merging `[revsets]` and `[revset-aliases]` so it's probably best to use a different section for things that are not revsets

File: cli/tests/test_rebase_command.rs [hunk 4]
```diff
@@ -496,6 +498,32 @@ fn test_rebase_multiple_destinations() {
...
+    let (stdout, stderr) = test_env.jj_cmd_ok(
+        &repo_path,
+        &[
+            "rebase",
+            "--config-toml=revsets.always-allow-large-revsets=true",
+            "-r",
+            "a",
+            "-d",
+            "b|c",
```
Comment (martinvonz) [inline cli/tests/test_rebase_command.rs:513]: optional: use `"-r=a"` and `"-d=b|c"` to save a bit of space

File: cli/src/config-schema.json (hunk 1 above)
Comment (thoughtpolice) [reply cli/src/config-schema.json:340]: FWIW, the name "large" comes from the fact that before the `all:` prefix existed (last year), you used the `--allow-large-revsets`/`-L` flag to `rebase` instead. See 9c8d8b73. So, it's really just bringing that name back from the grave. I think `singleton` is a bit too wordy, tbh.

File: cli/tests/test_rebase_command.rs [hunk 5]
```diff
@@ -496,6 +498,32 @@ fn test_rebase_multiple_destinations() {
...
+        ],
+    );
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @"");
```
Comment (martinvonz) [inline cli/tests/test_rebase_command.rs:517]: i'm not sure we need more coverage of the output of `jj rebase` here, but at least this is in `test_rebase_command.rs`, so it makes some sense. up to you

**Review by martinvonz — COMMENTED** [review]
I'm also not sure "large" is the best word but I'm not sure if "non-singleton" is good either. While most programmers know what "singleton" means, I'm a bit worried it's unnecessarily technical.

File: cli/src/config-schema.json (hunk 1 above)
Comment (noahmayr) [reply cli/src/config-schema.json:340]: Makes sense, and I agree it's wordy/technical. We can bikeshed the wording some other time when we come up with a better term

File: cli/tests/test_rebase_command.rs (hunk 5 above)
Comment (thoughtpolice) [reply cli/tests/test_rebase_command.rs:517]: It was basically just a copy/paste job, but I think removing it makes sense.

File: cli/src/config/misc.toml (hunk 3 above)
Comment (thoughtpolice) [reply cli/src/config/misc.toml:8]: I actually put it under `core.` at first, but then put it under `revsets.` since it seemed more "topical." Do you think `core.` is a better namespace?

Comment (martinvonz) [reply cli/src/config/misc.toml:8]: `ui.` works for me if we don't have a better place. it's specific to the UI, but so are almost all configs (`user.name` and `user.email` being good examples of non-UI configs)

Comment (thoughtpolice) [reply cli/src/config/misc.toml]: Migrated, thanks

**Review by martinvonz — APPROVED** [review]
